### PR TITLE
Get API host from the request's context

### PR DESCRIPTION
**Pull request by lubosmj**
closes #7385

### CODE REVIEW

**Comment by lubosmj** [comment]
The tests are failing because one part of the issue was addressed in pulpcore. Refer to https://github.com/pulp/pulpcore/pull/968 to learn more.

**Comment by mdellweg** [comment]
Depends on https://github.com/pulp/pulpcore/pull/968
Should we mark it as WIP until this has been merged?

File: pulp_container/app/serializers.py
Comment (ipanova) [inline pulp_container/app/serializers.py:112]: :+1:

**Comment by ipanova** [comment]
changes look good to me, let's wait on the core pr to me merged.

File: pulp_container/app/serializers.py
Comment (dkliban) [reply pulp_container/app/serializers.py:112]: We need to provide two code paths here so that the plugin can stay compatible with older versions of pulpcore that don't pass in this request object. We should file a task to remove the support for the old behavior once we no longer support pulpcore 3.8.

**Comment by lubosmj** [comment]
Tests are now passing because the additional change was merge in pulpcore. I believe we should now continue with the @dkliban's proposal.

File: pulp_container/app/serializers.py
Comment (dkliban) [reply pulp_container/app/serializers.py:112]: After some more discussion on IRC, we decided that it does not make sense to have the two code paths because the bug would still be present until pulpcore 3.9 is released. The other option is to release the pulpcore change as part of 3.8.1. If that does not pose any problems for other plugins, we will merge this PR after 3.8.1 is released.

Comment (mdellweg) [reply pulp_container/app/serializers.py:112]: Isn' it the other way around, and this change should bump the minimal supported pulpcore version to 3.8.0-dev?
And therefore only be released after 3.8?

Comment (ipanova) [reply pulp_container/app/serializers.py:112]: @mdellweg that's correct, we would need to raise the pulpcore requirement after 3.8.1 gets released

Comment (mdellweg) [reply pulp_container/app/serializers.py:112]: So i think, we need to raise that lower bound to 3.9(.dev?) with this PR. So that attempts to install it from source with the wrong version of pulpcore will fail already.
(we can drop the `dev` in the requirement on release)

**Comment by mdellweg** [comment]
In the container meeting we discussed, that this PR should bump the core requirement to 3.9.dev.

File: requirements.txt
Comment (lubosmj) [inline requirements.txt:1]: @mdellweg, I bumped the requirement here. This is what you wanted to see?

File: requirements.txt
Comment (lubosmj) [inline requirements.txt:1]: @mdellweg, I bumped the requirement here. This is what you wanted to see?

**Comment by ipanova** [comment]
@lubosmj can you make sure ci is green?

**Comment by lubosmj** [comment]
@ipanova, sorry, Travis CI is now green.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**
